lib/browser/api/touch-bar.js

```diff
@@ -51,10 +51,22 @@ class TouchBar extends EventEmitter {
         item.child.ordereredItems.forEach(registerItem)
       }
     }
+
+    let hasOtherItemsProxy = false
     items.forEach((item) => {
       if (!(item instanceof TouchBarItem)) {
         throw new Error('Each item must be an instance of TouchBarItem')
       }
+
+      // only allow one OtherItemsProxy per touchbar
+      if (item.type === 'other_items_proxy') {
+        if (!hasOtherItemsProxy) {
+          hasOtherItemsProxy = true
+        } else {
+          throw new Error('Must only have one OtherItemsProxy per TouchBar')
```

Comment:
We need to ensure we throw this error before we call `registerIteam` (probably in a different loop) otherwise we're going to have dangling event listeners.